azext_iot/iothub/providers/state.py

```diff
@@ -0,0 +1,277 @@
+
+    def upload_device_identity(self, identity):
+        device_id = identity["deviceId"]
+        auth_type = identity["authenticationType"]
+        edge = identity["capabilities"]["iotEdge"]
+        status = identity["status"]
+        ptp = identity["x509Thumbprint"]["primaryThumbprint"]
+        stp = identity["x509Thumbprint"]["secondaryThumbprint"]
+
+        if "status_reason" in identity.keys():
+            status_reason = identity["statusReason"]
+        else:
+            status_reason = None
+
+        if(auth_type == "sas"):
+            pk = identity["symmetricKey"]["primaryKey"]
+            sk = identity["symmetricKey"]["secondaryKey"]
+
+            iot_device_create(self.cmd, device_id, self.hub_name, edge, primary_key=pk, secondary_key=sk, status=status, \
+                status_reason=status_reason, resource_group_name=self.rg, auth_type_dataplane=self.auth_type)
+
+        elif(auth_type == "selfSigned"):
+            iot_device_create(self.cmd, device_id, self.hub_name, edge, auth_method='x509_thumbprint', \
+                    primary_thumbprint=ptp, secondary_thumbprint=stp, status=status, status_reason=status_reason, resource_group_name=self.rg, \
+                    auth_type_dataplane=self.auth_type)
+
+        elif(auth_type == "certificateAuthority"):
+                iot_device_create(self.cmd, device_id, self.hub_name, edge, auth_method='x509_ca', status=status, status_reason=status_reason, \
+                    resource_group_name=self.rg, auth_type_dataplane=self.auth_type)
+
+        else: 
+            logger.warning("Authorization type for device '{0}' not recognized.".format(device_id))
+
+    def upload_module_identity(self, identity):
+
+        device_id = identity["device_id"]
+        module_id = identity["module_id"]
+        auth_type = identity["authenticationType"]
+
+        if(auth_type == "sas"):
+            pk = identity["symmetricKey"]["primaryKey"]
+            sk = identity["symmetricKey"]["secondaryKey"]
+
+            iot_device_module_create(self.cmd, device_id, module_id, self.hub_name, primary_key=pk, secondary_key=sk, resource_group_name=self.rg, auth_type_dataplane=self.auth_type)
+
+        elif(auth_type == "selfSigned"):
+            ptp = identity["x509Thumbprint"]["primaryThumbprint"]
+            stp = identity["x509Thumbprint"]["secondaryThumbprint"]
+
+            iot_device_module_create(self.cmd, device_id, module_id, self.hub_name, auth_method='x509_thumbprint', primary_thumbprint=ptp, secondary_thumbprint=stp, \
+                resource_group_name=self.rg, auth_type_dataplane=self.auth_type)
+
+        elif(auth_type == "certificateAuthority"):
+
+            iot_device_module_create(self.cmd, device_id, module_id, self.hub_name, auth_method='x509_ca', resource_group_name=self.rg, auth_type_dataplane=self.auth_type)
+
+        else:
+            logger.warning("Authorization type for module '{0}' in device '{1}' not recognized.".format(module_id, device_id))
```

Comment:
Should these be logger.error, since the operation will fail for invalid auth_type